pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,339 @@
+				default:
+					usageError(cmd, "Wrong output format specified: %s", output)
+				}
+			}
+		},
+	}
+
+	cmd.Flags().Bool("all", false, "Build dependency trees for all image repositories")
+	cmd.Flags().StringP("output", "o", "json", "Output format of dependency tree(s)")
+	return cmd
+}
+
+// getRepos iterates over a given set of build configurations
+// and extracts all the image repositories on which other
+// image repositories are based
+func getRepos(configs []buildapi.BuildConfig, defaultNamespace string) map[string]struct{} {
+	avoidDuplicates := make(map[string]struct{})
+	for _, cfg := range configs {
+		for _, tr := range cfg.Triggers {
+			if tr.ImageChange != nil {
+				switch tr.ImageChange.From.Namespace {
+				case "":
+					avoidDuplicates[defaultNamespace+"/"+tr.ImageChange.From.Name] = struct{}{}
+				default:
+					avoidDuplicates[tr.ImageChange.From.Namespace+"/"+tr.ImageChange.From.Name] = struct{}{}
+				}
+			}
+		}
+	}
+
+	return avoidDuplicates
+}
+
+// parse accepts the input passed by the user and
+// breaks it down to the desired values if possible
+func parse(input string) (string, string, error) {
+	args := strings.Split(input, ":")
+	switch len(args) {
+	case 1:
+		return args[0], "latest", nil
+	case 2:
+		if strings.TrimSpace(args[1]) == "" {
+			return args[0], "latest", nil
+		}
+		return args[0], args[1], nil
+	default:
+		return "", "", invalidRepoTagErr
+	}
+}
+
+// findRepoDeps accepts an image repository name and a list of build
+// configurations and returns the dependency tree of the specified
+// image repository
+func findRepoDeps(namespace, name, tag string, all bool, buildConfigList []buildapi.BuildConfig) *ImageRepo {
+	root := &ImageRepo{
+		Name:      name,
+		Namespace: namespace,
+		Tags:      []string{tag},
+	}
+
+	// Search all build configurations in order to find the image
+	// repositories depending on the specified image repository
+	var outputName, outputTag string
+	for _, cfg := range buildConfigList {
+		for _, tr := range cfg.Triggers {
+			// If an ImageChangeTrigger has been specified (ie. it's not nil), use its From
+			// and Tag fields to compare against the specified image repository
+			if tr.ImageChange != nil && name == tr.ImageChange.From.Name && (all || tag == tr.ImageChange.Tag) {
+				// Either To & Tag or DockerImageReference will be used as output
+				if cfg.Parameters.Output.To != nil && cfg.Parameters.Output.To.Name != "" {
+					outputName = cfg.Parameters.Output.To.Name
+					outputTag = cfg.Parameters.Output.Tag
+					if cfg.Parameters.Output.To.Namespace != "" {
+						namespace = cfg.Parameters.Output.To.Namespace
+					}
+				} else {
+					ref, err := imageapi.ParseDockerImageReference(cfg.Parameters.Output.DockerImageReference)
+					checkErr(err)
+					outputName = ref.Name
+					outputTag = ref.Tag
+				}
+
+				// Build all children and their dependency trees recursively
+				child := findRepoDeps(namespace, outputName, outputTag, all, buildConfigList)
```

Comment:
namespace here should be either cfg.Parameters.Output.To.Namespace(as you have) or cfg.Namespace.  As you have it now, the second case is going to use the previous imageRepo's namespace and not the buildConfig's namespace, which would be incorrect if the buildConfig is in a different namespace than the imagerepo that triggered the buildConfig.